### ISSUE TITLE
onChange event for MentionsInput component captures arrow keys, but not mouse clicks

### ISSUE THREAD

**Issue by mitchizzle** [signavio/react-mentions#410]
**Steps to reproduce:**

Using a very simple example of the MentionsInput component, with a simple onChange function instead of the recompose handlers from the examples.

This works fine for keyboard events (trigger with '@', and navigate suggestions list with arrow keys + enter). However clicking on suggestions in the list simply closes the list, rather than selecting the clicked item. And the function supplied to onChange is never called for click events.

Is there a good way to enable click events through a simple onChange function, without importing the 'recompose' handlers from the examples?

```
class App extends Component {
  
  constructor(){
    super()
    this.state = {
      aValue: 'beep beep',
      users: [ {id: 'walter',display: 'Walter White',}, {id: 'jesse', display: 'Jesse Pinkman',}, {id: 'gus', display: 'Gustavo "Gus" Fring',}, ]
    }
  }
  
  handleChange(event){
    console.log('change event did run')
    this.setState({
      aValue: event.target.value,
    });
  }

  render() {
    return (
      <div className="App">
        <MentionsInput
          value={this.state.aValue}
          placeholder={"Mention people using '@'"}
          onChange={this.handleChange.bind(this)}
          style={styleBlock}
        >
          <Mention data={ this.state.users } />

        </MentionsInput>
      </div>
    );
  }
}
```

**Expected behaviour:**
Clicking on a suggestion will select that suggestion, or at least fire onChange event

**Observed behaviour:**
Clicking on a suggestion closes the suggestionlist without firing onChange event

**Workaround:**
None

**Comment by mitchizzle** [comment]
**Workaround:**
Falling back to _react-mentions_ version 3.3.2 (instead of latest version=4.0.0) restores selection by mouse, without any modifications to code above

**Comment by raelmiu** [comment]
Had the same issue. Also went back to 3.2.2. Waiting for a fix.

**Comment by claritypsychology** [comment]
Also having this issue.

**Comment by jfschwarz** [comment]
a fix has been released with `v4.0.1`

**Comment by AlimovSV** [comment]
Looks like this bug introduced again in 4.0.2
